client/src/components/Homepage.js

```diff
@@ -18,40 +19,91 @@ function Homepage({ searchText }) {
+              <path
+                fillRule='evenodd'
+                d='M14.707 12.707a1 1 0 01-1.414 0L10 9.414l-3.293 3.293a1 1 0 01-1.414-1.414l4-4a1 1 0 011.414 0l4 4a1 1 0 010 1.414z'
+                clipRule='evenodd'
+              />
+            </svg>
           )}
-          <div className="flex flex-col space-y-2 p-5 py-3.5">
-            <div className="flex justify-between items-center">
-              <h2 className="text-2xl font-medium text-black">{post.title}</h2>
-              <div>
-                <PostMenu />
+        </button>
+      </div>
+      <form className='space-y-8'>
+        {feed.map((post) => (
+          <div
+            key={post.id}
+            className='flex rounded-xl border border-gray-300 max-h-60 overflow-hidden cursor-pointer'
+            onClick={() => console.log('TODO: Load this Post!')}
+          >
+            {post.image && <img className='h-full w-60' src={post.image} alt='Logo' />}
+            <div className='flex flex-col space-y-2 p-5 py-3.5'>
+              <div className='flex justify-between items-center'>
+                <h2 className='text-2xl font-medium text-black'>{post.title}</h2>
+                <div>
+                  <PostMenu />
+                </div>
               </div>
+              <p className='text-sm font-medium text-black'>
+                By{' '}
+                <span className='font-extrabold text-transparent bg-clip-text bg-gradient-to-b from-red-500 to-purple-400'>
+                  @{post.User.name}
+                </span>
+              </p>
+              <p className='text-base text-black overflow-hidden'>{post.body}</p>
             </div>
-            <p className="text-sm font-medium text-black">
-              By{" "}
-              <span className="font-extrabold text-transparent bg-clip-text bg-gradient-to-b from-red-500 to-purple-400">
-                @{post.userId}
-              </span>
-            </p>
-            <p className="text-base text-black overflow-hidden">{post.body}</p>
           </div>
-        </div>
-      ))}
-    </form>
+        ))}
+      </form>
+    </div>
   );
 }
 
+function useUpdateCheck(feed, searchText, sortOrder, delay) {
+  useEffect(() => {
+    const feedURL = `${process.env.REACT_APP_HOST || ''}/feed/get-feed`;
+
+    function checkFeed() {
+      console.log('Checking for feed updates 👀');
+      let searchParams = '?' + new URLSearchParams({ searchText, sortOrder });
+      axios
+        .get(feedURL + searchParams)
+        .then((response) => {
+          let result = response.data;
+          if (JSON.stringify(result) !== JSON.stringify(feed)) {
+            alert('AHHHHHHHHHHHHHH REFRESH THE PAGE NOWWWWWW');
+          }
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    }
+
+    let id = setInterval(checkFeed, delay);
+    return () => clearInterval(id);
+  }, [feed, searchText, sortOrder, delay]);
+}
+
```

Comment:
Does this need to be a separate function outside of the homepage function? If it does then it should probably be above the homepage one for ease of understanding the file

Reply:
This is a [custom hook](https://reactjs.org/docs/hooks-custom.html), which I think will only be used here. If we want to use it across multiple components, I think we can just extract it to a separate file. I can definitely move it to the top of this file for readability though for now though.

Reply:
I ended up revamping this to be a simple `useEffect`, as it wasn't providing any additional benefits to make a custom hook.